### ISSUE TITLE
Option to disable SourceIPVerification while chaining CNI

### ISSUE THREAD

**Issue by CiraciNicolo** [cilium/cilium#33889]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

higher than v1.15.7 and lower than v1.16.0

### What happened?

While using Cilium in chain with KubeOVN, using Hubble, I detected that Cilium drops packets that are handled by KubeOVN. In particular this packets are generated by the VPC Nat GW, implemented to permit custom VPC - inside KubeOVN - to reach external networks.

Looking around I stumbled upon "SourceIPVerification" that, name intended, should skip the src ip validation and let the packet continue. Enabling that flag, unfortunately, does not work.

### How can we reproduce the issue?

1. Install KubeOVN with Cilium as instructed here: https://kubeovn.github.io/docs/v1.12.x/en/advance/with-cilium/
2. Create VPC
3. Create NatGW
4. ping any external ip such as 1.1.1.1

### Cilium Version

1.15.6

### Kernel Version

5.15.0-116-generic

### Kubernetes Version

1.29.6

### Cilium Users Document

- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by CiraciNicolo** [comment]
Searching around I stumbled upon `disable-sip-verification`, but still no luck

**Comment by CiraciNicolo** [comment]
Ok, it seems that `SourceIPVerification` is not persistent. Re-enabling that actually works, there is a way to make it persistant?

**Comment by squeed** [comment]
@CiraciNicolo by not persistent, you mean you are setting this with `cilium-dbg config` directly in the running process? Funny.

**Comment by CiraciNicolo** [comment]
Yes, I mean exactly that. The settings is not persistent to a rollout restarts. Looking around I found what seems to be an env variable, but I was hoping someone would be able to guide me through the setting ☺️

**Comment by squeed** [comment]
It does not seem like there currently exists the ability to plumb this through; it is hard-coded to true. This will take a code change to add a configuration flag.

A good first PR :-)

**Comment by squeed** [comment]
PR #23985 added the ability to do so at endpoint creation time, but that's awkward for most users. Better would be a daemon flag or the ability to do so via CNI configuration file.

**Comment by CiraciNicolo** [comment]
I was already pocking the code, I was hoping to discuss it first ☺️ trying to understand why the value is hardcoded as "true". I'm on mobile now, I will link the loc asap

**Comment by squeed** [comment]
Aha. Perhaps @oblazek knows the history behind that?

**Comment by CiraciNicolo** [comment]
I'm referring to this: https://github.com/cilium/cilium/blob/0f32f68f1c0c5fe13136ee438cf2a8d3e76333b9/daemon/cmd/daemon_main.go#L1348

Other options are configurable, but this - as well as others - is hardcoded. Probably the best course of action is to make everything configurable. Will PR this ASAP 🤗

**Comment by oblazek** [comment]
Hey, yeah imo there is no reason why this shouldn't be configurable as long as the current behavior works. We are using endpoint API directly, that's why we were fine with the hardcoded "true".

**Comment by xvzf** [comment]
The PR from @CiraciNicolo looks fairly simple - any idea when this will be released by chance? I just ran into this after hours of tcpdump (that's on me) and the workaround setting it after startup is fairly hacky

**Comment by log1cb0mb** [comment]
I need this too bad!

**Comment by CiraciNicolo** [comment]
Hi! I will work on this in the next couple of weeks.

**Comment by SkalaNetworks** [comment]
Linking the issue from Kube-OVN where the current Cilium behaviour is blocking some features such as NAT gateways: https://github.com/kubeovn/kube-ovn/issues/4314

Are we waiting on something for the merge? I'm thinking it could be enhanced further in a separate PR to disable SIP on a per-pod basis.

**Comment by Sha0** [comment]
For anyone else experiencing the same challenge with **Cilium v1.10.6** (and possibly other versions), this hack works for me:

`kubectl -n kube-system patch daemonsets cilium --type=json --patch='[{"op":"replace","path":"/spec/template/spec/containers/0/args","value":["-c","sed -i '"'"'s/ifndef DISABLE_SIP_VERIFICATION/ifdef DISABLE_SIP_VERIFICATION/'"'"' /var/lib/cilium/bpf/lib/lxc.h; exec cilium-agent --config-dir=/tmp/cilium/config-map"]},{"op":"replace","path":"/spec/template/spec/containers/0/command","value":["sh"]}]'`

What does it mean?  The Cilium agent **DaemonSet** will be adjusted so that a **sed** command first adjusts a C header-file, then **exec** will chain to the usual **cilium-agent** binary, which will then compile the BPF; including disabling the source IPv4 address verification.

This permits usage such as:
- IPv4 spoofing from a pod
- **iptables** NAT inside a pod

The **kube-proxy** and **externalTrafficPolicy: Cluster** have always presented challenges for _preserving source IPv4 addresses, such as in log-files_ and using NAT and routing can be part of an answer.

**Comment by CiraciNicolo** [comment]
Your solution, in my opinion, is too much hacky. Our current workaround is to use a sidecar container that launches `cilium config SourceIPVerification=Disabled`, just for sharing 😊

**Comment by Sha0** [comment]
Yes, it is very hacky.  Unfortunately, **SourceIPVerification** is not available in **v1.10.6**.
